src/Standards/Generic/Sniffs/CodeAnalysis/UselessOverridingMethodSniff.php

```diff
@@ -56,7 +56,15 @@ public function process(File $phpcsFile, $stackPtr)
         $token  = $tokens[$stackPtr];
 
         // Skip function without body.
-        if (isset($token['scope_opener']) === false) {
+        if (isset($token['scope_opener'], $token['scope_closer']) === false) {
+            return;
+        }
+
+        $conditions    = $token['conditions'];
+        $lastCondition = end($conditions);
+
+        // Skip functions that are not a method part of a class, anon class or trait.
+        if (in_array($lastCondition, [T_CLASS, T_ANON_CLASS, T_TRAIT]) === false) {
```

Comment:
Nitpick: as this is a fixed set of tokens anyway, can this array be declared in a class property and checked against via `isset()` ?

This would also prevent potential bugs due to `in_array()` in the code as-is being used without the `$strict` condition. (let me know if you want me to explain that one in more detail)

Reply:
> Nitpick: as this is a fixed set of tokens anyway, can this array be declared in a class property and checked against via isset() ?

I gave this a try in 73ff2b1c8ced349faf5b3a59e7dbcb926f5feb88. Is this more or less what you had in mind? I'm not super happy with the name of the class property. Please me know if you have a better suggestion.

> This would also prevent potential bugs due to in_array() in the code as-is being used without the $strict condition. (let me know if you want me to explain that one in more detail)

Good point about the lack of `$strict` when calling `in_array()`. I'm assuming you are talking about the possibility of an in_array() behaving unexpectedly as without `$strict` `in_array()` performs a loose comparison, but please let me know if you have something else in mind.

Reply:
> I gave this a try in [73ff2b1](https://github.com/PHPCSStandards/PHP_CodeSniffer/commit/73ff2b1c8ced349faf5b3a59e7dbcb926f5feb88). Is this more or less what you had in mind?

Is it 👍🏻 

> I'm not super happy with the name of the class property. Please let me know if you have a better suggestion.

What about `$validOOscopes` or even just `$ooScopes` ? (as well as changing the property description to something describing that, like "OO scopes in which a call to `parent::method()` can exist")

> > This would also prevent potential bugs due to in_array() in the code as-is being used without the $strict condition. (let me know if you want me to explain that one in more detail)
> 
> Good point about the lack of `$strict` when calling `in_array()`. I'm assuming you are talking about the possibility of an in_array() behaving unexpectedly as without `$strict` `in_array()` performs a loose comparison, but please let me know if you have something else in mind.

Correct, in particular as PHP native token constants will always be ints, but polyfilled tokens and PHPCS native tokens (like `T_ANON_CLASS`) will always be strings and the strings would be juggled to an int without the `$strict` parameter being `true`.

Reply:
> What about $validOOscopes or even just $ooScopes ? (as well as changing the property description to something describing that, like "OO scopes in which a call to parent::method() can exist")

Thanks! I went with `$validOOScopes` to hint that the property is not about all the OO scopes.